RELEASE-NOTES.md

```diff
@@ -1,5 +1,40 @@
 # Prefect Release Notes
 
+## Release 2.16.1
+
+### TODO: !Highlight!
+
+### Enhancements
+- Add filter to prevent logging the current API key — https://github.com/PrefectHQ/prefect/pull/12072
+- Add artifact data to flow run graph API — https://github.com/PrefectHQ/prefect/pull/12105
+- Update `flow.serve` to support multiple schedules — https://github.com/PrefectHQ/prefect/pull/12107
```

Comment:
@bunchesofdonald if you make a highlight, would be nice if you can move to highlight section with 

```
For implementation details, see the following pull request:
    - https://github.com/PrefectHQ/prefect/pull/12107
 ```